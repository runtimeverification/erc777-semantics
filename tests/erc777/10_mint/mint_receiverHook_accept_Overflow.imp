test.preamble();

// operatorMint() through default operator (unless otherwise specified), operatorMint() from caller.
// receiver hook, always accepting, minting to overflow totalSupply

test.registerTokensReceivedHook(5, 1); // Hook addresses 1 and 2 are accepting.
test.makeCaller(5);
test.operatorMint(5, 115792089237316195423570985008687907853269984665640564039457584007913129639935, "", ""); //from caller

test.makeCaller(8);
test.operatorMint(5, 115792089237316195423570985008687907853269984665640564039457584007913129639935, "", ""); //through default operator
