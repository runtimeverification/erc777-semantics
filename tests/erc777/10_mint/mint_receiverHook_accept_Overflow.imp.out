Registering receiver 5 with hook 1
Changing the caller to 5
5 is minting 115792089237316195423570985008687907853269984665640564039457584007913129639935 for 5 (data:""),  (operatorData:"")
REVERT
Balance of 5 is 50
Total supply is 550
Changing the caller to 8
8 is minting 115792089237316195423570985008687907853269984665640564039457584007913129639935 for 5 (data:""),  (operatorData:"")
REVERT
Balance of 5 is 50
Total supply is 550
